modules/extensions-support/src/main/java/org/mule/module/extension/internal/manager/DefaultExtensionManager.java

```diff
@@ -0,0 +1,425 @@
+        LOGGER.info("Discovered {} extensions", discovered.size());
+
+        ImmutableList.Builder<Extension> accepted = ImmutableList.builder();
+
+        for (Extension extension : discovered)
+        {
+            if (registerExtension(extension))
+            {
+                accepted.add(extension);
+            }
+        }
+
+        return accepted.build();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean registerExtension(Extension extension)
+    {
+        LOGGER.info("Registering extension {} (version {})", extension.getName(), extension.getVersion());
+        final String extensionName = extension.getName();
+
+        if (register.containsExtension(extensionName))
+        {
+            return maybeUpdateExtension(extension, extensionName);
+        }
+        else
+        {
+            doRegisterExtension(extension, extensionName);
+            return true;
+        }
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public <C> void registerConfigurationInstanceProvider(String providerName, ConfigurationInstanceProvider<C> configurationInstanceProvider)
+    {
+        Configuration configuration = configurationInstanceProvider.getConfiguration();
+        ExtensionStateTracker extensionStateTracker = register.getExtensionState(configuration);
+        extensionStateTracker.registerConfigurationInstanceProvider(configuration, providerName, configurationInstanceProvider);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public <C> C getConfigurationInstance(final ConfigurationInstanceProvider<C> configurationInstanceProvider, OperationContext operationContext)
+    {
+        return configurationInstanceProvider.get(operationContext, new ConfigurationInstanceRegistrationCallback()
+        {
+            @Override
+            public <C> void registerNewConfigurationInstance(ConfigurationInstanceProvider<C> configurationInstanceProvider, C configurationInstance)
+            {
+                registerConfigurationInstance(configurationInstanceProvider.getConfiguration(),
+                                              configurationInstanceProvider.getName(),
+                                              configurationInstance);
+            }
+        });
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public OperationExecutor getOperationExecutor(OperationContext operationContext)
+    {
+        Extension extension = register.getExtension(operationContext.getOperation());
+        Map<String, ConfigurationInstanceProvider> providers = register.getConfigurationInstanceProviders(extension);
+
+        int matches = providers.size();
+
+        if (matches == 1)
+        {
+            ConfigurationInstanceProvider<Object> provider = providers.values().iterator().next();
+            return getOperationExecutor(provider, operationContext);
+        }
+        else if (matches > 1)
```

Comment:
Doesn't make sense to. The upcoming change on OperationExecutor's lifecycle will cause this method to be gone